osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs

```diff
@@ -373,7 +373,9 @@ private void handleTimingPoint(string line)
             string[] split = line.Split(',');
 
             double time = getOffsetTime(Parsing.ParseDouble(split[0].Trim()));
-            double beatLength = Parsing.ParseDouble(split[1].Trim());
+            double beatLength = Parsing.ParseDouble(split[1].Trim(), allowNaN: true);
```

Comment:
Add a comment before this line explaining why the nan allowance is here (and pointing to the relevant code which handles this in `LegacyDecoder`).